src/rust/kafka-executor/src/event_handler.rs

```diff
@@ -0,0 +1,13 @@
+use crate::errors::CheckedError;
+
+#[async_trait::async_trait]
+pub trait EventHandler {
+    type InputEvent;
+    type OutputEvent;
+    type Error: CheckedError;
+
+    async fn handle_event(
```

Comment:
This isn't really a generic event handler. In Kafka we have three types of clients:
1. Producer
2. Consumer
3. Stream processor

This appears to be a client of type (3).